openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1191,29 +1193,120 @@ Cache-Control: no-store
 }
 ```
 
+# Notification Endpoint {#notification_endpoint}
+
+This endpoint is used by the Wallet to notify the Credential Issuer whether a Credential has been successfully received or not. It enables the Credential Issuer to take subsequent actions after issuance, depending on whether the Credential has been accepted and successfully stored by the Wallet, rejected by the Wallet, or errors or other unforeseen circumstances have occurred during the Wallet's processing. The Credential Issuer needs to return the `notification_id` in the Credential Response or a Batch Credential Response for the Wallet to be able to use this Endpoint. Support for this endpoint is OPTIONAL.
+
+The Wallet MUST present to the Notification Endpoint a valid Access Token issued at the Token Endpoint as defined in (#token_endpoint). 
+
+Note: A Credential Issuer that requires a request to the Notification Endpoint MUST ensure the Access Token issued by the Authorization Server is valid at the Notification Endpoint.
+
+The notification from the Wallet is idempotent. The Credential Issuer MUST return success even if it receives multiple identical calls from the Wallet for the same `notification_id`s.
+
+The Wallet MAY retry if the call to this endpoint fails for a temporary reason. The Credential Issuer SHOULD pre-determine the amount of time within which it expects the notification. Therefore, even a well-formed notification from the Wallet could fail if received by the Credential Issuer after this time period. The Credential Issuer may not receive the notification, meaning it is unknown whether the Wallet successfully stored the credential or not - it is left to the Credential Issuer to decide how to proceed in this case.
```

Comment:
```suggestion
The Wallet MAY retry if the call to this endpoint fails.
```

1. Its hard to qualify what constitutes a temporary reason in this context
2. Some notifications likely wont have a known amount of time to expect a notification within so I think we should just remove this.

Reply:
Temporary might be a relatively well understood thing - dns fails, tcp connect timeouts, http 503, etc.

We could say the opposite, that the client shouldn't retry if it gets a permanent failure. I'd say we shouldn't imply the client should retry in the face of a known permanent failure (like a http 400 bad request or 410 gone).

Reply:
yeah, i am not ok removing these sentences entirely... just removing `for a temporary reason` would be ok.